driver-async/src/main/com/mongodb/async/client/ClientSessionImpl.java

```diff
@@ -31,9 +31,15 @@
 
 class ClientSessionImpl extends BaseClientSessionImpl implements ClientSession {
 
+    private enum TransactionState {
+        NONE, IN, DONE, ABORTED
```

Comment:
The spec has a STARTED state as well but I don't see any need for it yet in the implementation.  Open question on the spec about whether it needs to be in the spec.

Reply:
No sure what the difference between STARTED and IN is

Reply:
The started state is not needed but I think it makes sense to have it in the diagram to address the no-op abort/commit behavior.